File: Core/Concepts/Entities/BikeStation.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Core.Concepts.Entities
{

    public class BikeStation
    {
        [Key]
        public int StationId { get; set; }

        [Required, StringLength(100)]
        public string StationName { get; set; } = string.Empty;

        [Required]
        [Column(TypeName ="decimal(9,6)")]
        public double Lat { get; set; }

        [Required]
        [Column(TypeName ="decimal(9,6)")]
        public double Lon { get; set; }
		public List<BikeStationDock> BikeStationDocks { get; set; }

        [NotMapped]
        public int AvailableDocks { get; set; }
        [NotMapped]
        public int TotalDocks { get; set; }
    }

}

File: Web/wwwroot/scripts/secure/BikeStation.ts
import Dock from "@/scripts/Dock";

export default class BikeStation{
	StationId: number;
	Lat: number;
	Lon: number;
	BikeStationDocks : Dock[];
	TotalDocks: number;
	AvailableDocks: number;
	StationName: string;


	constructor(lat: number, lon: number, docks: Dock[], totalDock: number, availableDocks: number, StationName : string, StationId : number) {
		this.Lat = lat;
		this.Lon = lon;
		this.BikeStationDocks = docks;
		this.TotalDocks = totalDock;
		this.AvailableDocks = availableDocks;
		this.StationName = StationName;
		this.StationId = StationId;
	}





}
